internal/tiltfile/tiltfile_state.go

```diff
@@ -623,6 +623,13 @@ func (s *tiltfileState) assembleK8sV2() error {
 	}
 
 	for workload, opts := range s.k8sResourceOptions {
+		if opts.nonWorkload {
```

Comment:
this isn't a big deal at all but it feels kinda weird to me to make the resource, put it in the map, and immediately get it back out of the map. i might do something like
```
var r Resource
var ok bool
if opts.nonWorkload {
    r... := makeResource
    s.k8sByName[opts.newName] = r
} else if r, ok = s.k8sByName[workload]; !ok {
    return error("no resource found...)
}
    r.updateWithOpts(opts)
```

Reply:
Unfortunately without a larger refactor this will fail on non-workload-only-resources.

Reply:
aww :( womp. okay cool